components/builder-api/src/server/resources/origins.rs

```diff
@@ -213,7 +213,22 @@ fn create_origin(req: HttpRequest,
                                  default_package_visibility: &dpv, };
 
     match Origin::create(&new_origin, &*conn).map_err(Error::DieselError) {
-        Ok(origin) => HttpResponse::Created().json(origin),
+        Ok(origin) => {
+            match OriginAudit::audit(&OriginAudit { operation:      OriginOperation::OriginCreate,
+                                                    origin:         &body.0.name,
+                                                    target_object:  &body.0.name,
+                                                    requester_id:   session.get_id() as i64,
+                                                    requester_name: session.get_name(), },
```

Comment:
A point of style on this... When creating new types like this, I have often found it useful to keep the members private and instead expose a functional interface to them, rather than embedding their internal structure at all the call sites. You can pass in the data you need as parameters, which can give you both cleaner code at the call site, as well as the flexibility to rearrange the internal workings of your type later.

This usecase seems somewhat similar to the Supervisor's [event stream interface](https://github.com/habitat-sh/habitat/blob/master/components/sup/src/event.rs)... all the call sites in the broader Supervisor call functions like `service_stopped(service: &Service)`, while the details of creating and sending that event are encapsulated away within the `event` module itself.

Here, you could plausibly end up with something like `audit_origin_creation(&origin_name, &session)`, and hide everything else away.

It's certainly not a blocker, but it's an approach that I've found quite valuable in the past; maybe it could work here, as well.

Reply:
I like that! I was pondering a better way. I'll re-work it following your suggestion.

Reply:
Handled in https://github.com/habitat-sh/builder/pull/1310/commits/5ce40956fd473db4c9850791c206c3a1dca8829a